libc/test/UnitTest/LibcTestMain.cpp

```diff
@@ -43,7 +43,15 @@ TestOptions parseOptions(int argc, char **argv) {
 
 } // anonymous namespace
 
-extern "C" int main(int argc, char **argv, char **envp) {
+// The C++ standard forbids declaring the main function with a linkage specifier
+// outisde of 'freestanding' mode, only define the linkage for hermetic tests.
+#ifdef __STDC_HOSTED__
```

Comment:
Do you need to explicitly check if it's not 0?
```
$ diff <(clang++ -dM -E - < /dev/null) <(clang++ -dM -E -ffreestanding - < /dev/null)
86d85
< #define __GCC_HAVE_DWARF2_CFI_ASM 1
243c242
< #define __STDC_HOSTED__ 1
---
> #define __STDC_HOSTED__ 0
```

Reply:
You're right, just noticed once I got undefined references in the GPU tests. Just checking the value is how the clang headers do it.